src/public/modules/forms/services/FormFeedbackService.ts

```diff
@@ -0,0 +1,112 @@
+import axios from 'axios'
+import moment from 'moment-timezone'
+
+import { FormFeedbackPostDto } from '../../../../types'
+import {
+  FormFeedbackResponseDto,
+  GetFormFeedbackDto,
+} from '../../../../types/form_feedback'
+import { CsvGenerator } from '../helpers/CsvGenerator'
+
+// Exported for testing
+export const PUBLIC_FORM_ENDPOINT = '/api/v3/forms'
+export const ADMIN_FORM_ENDPOINT = '/api/v3/admin/forms'
+
+/**
+ * Post feedback for a given form.
+ * @param formId the id of the form to post feedback for
+ * @param feedbackToPost object containing the feedback
+ * @returns the posted feedback
+ */
+export const postFeedback = async (
+  formId: string,
+  feedbackToPost: FormFeedbackPostDto,
+): Promise<FormFeedbackResponseDto> => {
+  return axios
+    .post<FormFeedbackResponseDto>(
+      `${PUBLIC_FORM_ENDPOINT}/${formId}/feedback`,
+      feedbackToPost,
+    )
+    .then(({ data }) => data)
+}
+
+/**
+ * Retrieves feedback for a given form.
+ * @param formId the id of the form to retrieve feedback for
+ * @returns object containing average?, count, and feedback[]
+ */
+export const getFeedback = async (
+  formId: string,
+): Promise<GetFormFeedbackDto> => {
+  return axios
+    .get<GetFormFeedbackDto>(`${ADMIN_FORM_ENDPOINT}/${formId}/feedback`)
+    .then(({ data }) => data)
+}
+
+/**
+ * Counts feedback for a given form.
+ * @param formId the id of the form to count feedback for
+ * @returns the count of the retrieved feedback
+ */
+export const countFeedback = async (formId: string): Promise<number> => {
+  return axios
+    .get<number>(`${ADMIN_FORM_ENDPOINT}/${formId}/feedback/count`)
+    .then(({ data }) => data)
+}
+
+/**
+ * Class to encapsulate the FeedbackCsv and its attributes
+ * Exported for testing
+ */
+export class FeedbackCsvGenerator extends CsvGenerator {
```

Comment:
should we move this class into a separate file? all the other functions have to do with API request/response, so this seems unrelated

Reply:
shifted